Fix kIOMasterPortDefault deprecation on macOS 12 and later

The warning looks like this:

```
src/uart_enum_osx.c:49:47: warning: 'kIOMasterPortDefault' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]
    kernResult = IOServiceGetMatchingServices(kIOMasterPortDefault, classesToMatch, matchingServices);
                                              ^~~~~~~~~~~~~~~~~~~~
                                              kIOMainPortDefault
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/IOKit.framework/Headers/IOKitLib.h:123:19: note: 'kIOMasterPortDefault' has been explicitly marked deprecated here
const mach_port_t kIOMasterPortDefault
                  ^
1 warning generated.
```

--- a/src/uart_enum_osx.c
+++ b/src/uart_enum_osx.c
@@ -46,7 +46,7 @@ static kern_return_t FindModems(io_iterator_t *matchingServices)
                              CFSTR(kIOSerialBSDAllTypes));
     }
 
-    kernResult = IOServiceGetMatchingServices(kIOMasterPortDefault, classesToMatch, matchingServices);
+    kernResult = IOServiceGetMatchingServices(kIOMainPortDefault, classesToMatch, matchingServices);
 
     return kernResult;
 }
@@ -112,7 +112,6 @@ struct serial_info *find_serialports()
 
     io_service_t modemService;
     kernResult = KERN_FAILURE;
-    Boolean modemFound = false;
 
     // Initialize the returned path
     *bsdPath = '\0';
@@ -140,7 +139,6 @@ struct serial_info *find_serialports()
                 new_info->next = info;
                 info = new_info;
 
-                modemFound = true;
                 kernResult = KERN_SUCCESS;
 
                 io_service_t device = GetUsbDevice(modemService);

